topic-operator/src/main/java/io/strimzi/operator/topic/v2/BatchingTopicController.java

```diff
@@ -545,7 +558,8 @@ private void accumulateResults(Map<ReconcilableTopic, Either<TopicOperatorExcept
         var apparentlyDifferentRf = currentStatesOrError.ok().filter(pair -> {
             var reconcilableTopic = pair.getKey();
             var currentState = pair.getValue();
-            return currentState.uniqueReplicationFactor() != reconcilableTopic.kt().getSpec().getReplicas();
+            return reconcilableTopic.kt().getSpec().getReplicas() != null
+                && currentState.uniqueReplicationFactor() != reconcilableTopic.kt().getSpec().getReplicas();
```

Comment:
I'm not sure this is right. If the `spec.rf` is missing it means "broker default". That has a value even if it's not explicit. And so the actual number of replicas can differ from that.

Reply:
Let me check this one. Thanks.